qutebrowser/utils/log.py

```diff
@@ -526,6 +526,30 @@ def hide_qt_warning(pattern: str, logger: str = 'qt') -> typing.Iterator[None]:
         logger_obj.removeFilter(log_filter)
 
 
+def init_from_config(conf: typing.Any) -> None:
```

Comment:
`conf` is of type `qutebrowser.config.config.ConfigContainer`, but actually importing that creates a cyclic reference. I couldn't figure out the proper string to use as a forward reference either. I'm open to suggestions here.

Reply:
You can import it under a `if typing.TYPE_CHECKING:` block, and then use `'config.ConfigContainer'` as a string in the annotation.

See https://mypy.readthedocs.io/en/stable/common_issues.html#import-cycles